### ISSUE TITLE
Iterator for Data

### ISSUE THREAD

**Issue by vitreo12** [vitreo12/omni#73]
To allow this:

```nim
init:
    data = Data(10)
    for entry in data:
        print(entry)
```

and this:

```nim
struct Vector:
    x sig
    y sig
    z sig

init:
    dataLength = 100
    data = Data(dataLength, dataType=Vector)
    
    for entry in data:
        entry = Vector()
```

**Comment by vitreo12** [comment]
Added functionality for getting values... Setting them still requires normal looping

**Comment by vitreo12** [comment]
Implemented setter too #84 :
It's implemented in the typed parser itself.
This:
```nim
for entry in data:
   entry = Vector()
```
will be turned into
```nim
for index_1238123 in 0..data.len-1:
    data[index_1238123] = Vector()
```
